.azure-pipelines/live-functional.yml

```diff
@@ -0,0 +1,127 @@
+        # We maintain those resources in the test subscription available for the tests
+        - script: tools/c7n_azure/tests/templates/provision.sh --skip keyvault cost-management-export containerservice databricks
+          displayName: "Provision Azure Resources"
+          env:
+            AZURE_CLIENT_ID: $(azure-client-id)
+            AZURE_CLIENT_SECRET: $(azure-client-secret)
+
+        - script: C7N_TEST_RUN=true C7N_FUNCTIONAL=yes pytest -v -m "not skiplive" tools/c7n_azure/tests
+          displayName: "Run Azure tests without cassettes"
+
+        - script: tools/c7n_azure/tests/templates/cleanup.sh --skip keyvault cost-management-export containerservice databricks
+          displayName: "Cleanup Azure Resources"
+          condition: always()
+
+    - job: 'azure_nightly_functions_test'
+      displayName: 'Azure Functions Nightly Test'
+      timeoutInMinutes: 0
+      cancelTimeoutInMinutes: 0
+
+      pool:
+        vmImage: 'Ubuntu-16.04'
+
+      steps:
+        - checkout: self
+          fetchDepth: 1
+
+        - task: UsePythonVersion@0
+          inputs:
+            versionSpec: '3.7'
+            architecture: 'x64'
+
+        - script: python -m pip install --upgrade pip && pip install . && pip install -r requirements-dev.txt
+          displayName: "Install Dependencies"
+
+        - script: ./test_functions.sh
+          workingDirectory: tools/c7n_azure/tests/azure-functions
+          displayName: "Run Azure Functions Test"
+          env:
+            AZURE_CLIENT_ID: $(azure-client-id)
+            AZURE_CLIENT_SECRET: $(azure-client-secret)
+            AZURE_SUBSCRIPTION_ID: $(azure-subscription-id)
+            AZURE_TENANT_ID: $(azure-tenant-id)
+
+  # If all functional test stages complete successfully then
+  # push a docker image and tag `nightly`.
+  - stage: post_success
+    displayName: After Successful Functional Tests
+    dependsOn: azure_live_functional
+
+    jobs:
+      - job: 'push_docker_nightly'
+        displayName: 'Push Nightly Docker Image'
+
+        pool:
+          vmImage: 'Ubuntu-16.04'
+
+        steps:
+          # C7N
+          - task: Docker@2
+            displayName: Build and Push C7N Nightly
+            inputs:
+              command: buildAndPush
+              Dockerfile: Dockerfile
+              containerRegistry: dockerServiceConnection
+              repository: cloudcustodian/c7n
+              tags: |
+                nightly
+                $(Build.BuildNumber)-nightly
+
+          # Mailer
+          - task: Docker@2
+            displayName: Build and Push Mailer Nightly
+            inputs:
+              command: buildAndPush
+              Dockerfile: tools/c7n_mailer/Dockerfile
+              containerRegistry: dockerServiceConnection
+              repository: cloudcustodian/mailer
+              tags: |
+                nightly
+                $(Build.BuildNumber)-nightly
```

Comment:
what's the value for build number? per the proposal we were looking for a date string here

Reply:
Yeah it is date with an incrementing integer added to the end if this runs more than once per day, for example `20191010.1-nightly` 

I'm open to anything here though.  I was searching around on docker hub and found it isn't unusual to be very verbose on these specific build tags - some people stick the commit hash in there after the date too.

Reply:
the existing per commit builds are already published with the git sha, if we go back to the spec issue its sans nightly suffix for date tagging, afaics the builds will always have a .1 attached which is effectively an implementation detail. it would nice to be able to just have this unambiguously referenced, ie just add the date for the tag.

also noting that part of the reason for this was the instability of the pypi azure dep stack for installs on c7n_azure which is now also resolved (frozen dep set). I still think this is useful though to make docker usage more humane in between releases.

Reply:
Yeah I'll remove the postfix.

Reply:
Updated to `%Y%m%d` which means that if functionals passed today we'd have a build tagged:

```
nightly
20191016-nightly
```

Reply:
is the nightly suffix on the date string redundant? I'm mostly indifferent on it.

Reply:
It is but best I could tell this style was most common - when you go to a more specific tag it retains the higher level category - so if you switch to some dated nightly tag its still obvious by reading your image description that it is a specific nightly rather than some other dated build (betas/previews/snapshots/whatever else could theoretically exist)